lib/os/p4wq.c

```diff
@@ -70,6 +84,7 @@ static FUNC_NORETURN void p4wq_loop(void *p0, void *p1, void *p2)
 			w->thread = _current;
 			sys_dlist_append(&queue->active, &w->dlnode);
 			set_prio(_current, w);
+			thread_set_requeued(_current, false);
```

Comment:
nitpicking: I'm not a huge fan of functions, whose complete body is checking a condition like
```
void f(bool x)
{
	if (x)
		do_one_thing();
	else
		do_another_thing();
}
```
if they're ever only called with `x` being constant, i.e. as `f(true)` or `f(false)`. In such cases I find having two functions `f_true()` and `f_false()`. In this case we could have `thread_set_requeued()` and `thread_clear_requeued()` maybe as 2 macros.

Reply:
Ah, but you can't put it on the stack[1], because the stack is incoherent memory and the work item by definition needs to be shared between threads.  And making an internal copy would work, but would require a heap or mem_slab or some other allocator, which Zephyr doesn't assume to be present.  The idiom where the user manages memory for things like this is pervasive in our utilities.

I can change the API though, sure.  Honestly I like the single call for clarity and trust the compiler to eliminate the dead branch via inlining, but that's a taste thing.

[1] Not naively, anyway.  But you could do it in SOF by making sure it lives entirely within its own 64-byte cache line, munging the pointer to reference the uncached mapping, and then passing that to the queue.  But that would be CAVS-specific code, obviously.